From aad8993d3ad5b44967fd1c7c2e5dfea7439a70f2 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Wed, 14 Oct 2015 11:26:27 +0800
Subject: [PATCH 0653/5242] MLK-11700 video: mxsfb: correct the fb pan display
 ret value

commit  977bf877b3357802722eb829b90a7370d48fb7c6 from
https://source.codeaurora.org/external/imx/linux-imx.git

The fb_pan_display() return value should be corrected
according to the wait_for_completion_timeout() return
value.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
(cherry picked from commit 459d01cbda39e51029c55e583909ef645037b7f3)

Conflicts:
	drivers/video/mxsfb.c
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 550ec51..9020a6d 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -949,13 +949,13 @@ static int mxsfb_pan_display(struct fb_var_screeninfo *var,
 		host->base + LCDC_CTRL1 + REG_SET);
 
 	ret = wait_for_completion_timeout(&host->flip_complete, HZ / 2);
-	if (ret < 0) {
+	if (!ret) {
 		dev_err(fb_info->device,
 			"mxs wait for pan flip timeout\n");
-		ret = -ETIMEDOUT;
+		return -ETIMEDOUT;
 	}
 
-	return ret;
+	return 0;
 }
 
 static int mxsfb_mmap(struct fb_info *info, struct vm_area_struct *vma)
-- 
1.7.9.5

